src/rkopenmdao/runge_kutta_integrator.py

```diff
@@ -986,59 +1000,81 @@ def _compute_jacvec_fwd_run_steps(self):
     def _compute_jacvec_fwd_run_steps_preparation_phase(self):
         self._update_integration_control_step()
 
-    def _compute_jacvec_fwd_run_steps_time_integration_phase(self):
-        delta_t = self.options["integration_control"].delta_t
-        butcher_tableau: ButcherTableau = self.options["butcher_tableau"]
-        time = self.options["integration_control"].step_time
+    def stage_values_computation(self, stage):
         step = self.options["integration_control"].step
-        for stage in range(butcher_tableau.number_of_stages()):
-            if self.comm.rank == 0:
-                print(
-                    f"Starting stage {stage} of fwd-mode jacvec productin step {step}."
-                )
-            if stage != 0:
-                self._accumulated_stages = (
-                    self._runge_kutta_scheme.compute_accumulated_stages(
-                        stage, self._stage_cache
-                    )
+        time = self.options["integration_control"].step_time
+        if self.comm.rank == 0:
+            print(f"Starting stage {stage+1} of fwd-mode jacvec product in step {step}.")
+        if stage != 0:
+            self._accumulated_stages = (
+                self._runge_kutta_scheme.compute_accumulated_stages(
+                    stage, 
+                    self._stage_cache
                 )
-                self._accumulated_stage_perturbations = (
-                    self._runge_kutta_scheme.compute_accumulated_stage_perturbations(
-                        stage, self._stage_perturbations_cache
-                    )
+            )
+            self._accumulated_stage_perturbations = (
+                self._runge_kutta_scheme.compute_accumulated_stage_perturbations(
+                    stage, 
+                    self._stage_perturbations_cache
                 )
-            else:
-                self._accumulated_stages.fill(0.0)
-                self._accumulated_stage_perturbations.fill(0.0)
-            self._stage_cache[stage, :] = self._runge_kutta_scheme.compute_stage(
+            )
+        else:
+            self._accumulated_stages.fill(0.0)
+            self._accumulated_stage_perturbations.fill(0.0)
+        print(self._accumulated_stages, self._accumulated_stage_perturbations)
+        self._stage_computation(stage)
```

Comment:
I wouldn't reuse the `_stage_computation()`-function here. If you look at it, you will see that it redoes a lot of what is done here before (i.e. setting values in `integration_control`, and also signals that a stage in the time integration is finished. I would suggest to just use `self._stage_cache[stage, :] = self._runge_kutta_scheme.compute_stage(...)` as it was before.